### PR TITLE
Refactoring the Models subclasses

### LINKED ISSUE #155: Code Modularization: Models
--------
Issue Description
--------
##### What did you expect to see happen?

The code should be reorganized to split the sub-classes of the Models into separate files and place them each in their own directory with an appropriate __init__.py and a Factory.py file.

##### What did you see instead?

All of the subclasses are in a single file which can sometimes make it hard to orient oneself and find code snippets for a given class.

##### Do you have a suggested fix for the development team?

See Question 1.

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

No error associated.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by maljovec**
Refactoring the Models into a module. Adding a util function that will recursively search for all of the subclasses of a base class. This last addition is being added to the Factories that automatically generate their constituent dictionaries. Also fixing the docstring for the PostProcessors module.

--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

Closes #155 

##### What are the significant changes in functionality due to this change request?

These changes should have no significant change on the running of the code. This effort is only to reorganize/clean up the code, and should have insignificant change on the performance of the code.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/Models/Factory.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+__base = 'Model'
+__interFaceDict = {}
+
+for classObj in utils.getAllSubclasses(eval(__base)):
+  key = classObj.__name__
+  __interFaceDict[key] = classObj
+
+
+## Can this be removed?
+#__interFaceDict                   = (__interFaceDict.items()+CodeInterfaces.__interFaceDict.items()) #try to use this and remove the code interface
```
Comment (PaulTalbot-INL) [inline framework/Models/Factory.py]: We should probably get an answer to this before merging.

**Comment by maljovec** [comment]
@PaulTalbot-INL the change I requested from #168 should now be on this branch, and I am done actively developing on this branch.

**Comment by maljovec** [comment]
Sorry, the Mac test was giving different results because it was not running in interactive mode. I have updated the test file so that it should give the same results whether the UI is loaded or if it performs the fallback, but it should not have even run the `interactive_ui` test cases if it does not have PySide installed? For instance, in the `/tests/framework/PostProcessors/TopologicalPostProcessor/tests` file the line:
```
required_libraries = 'AMSC PySide'
```
seems to be ignored, or otherwise it could load PySide, but it failed to load PySide.QtGui or QtCore (which I have seen before).

**Comment by PaulTalbot-INL** [comment]
Reviewed the additional commits up to this point, and they're fine.  It looks like our Windows woes continue for the hdf5 test; is the `DET` test failing a new phenomenon?

**Pull request merged**